book/src/super-sql/types/named.md

```diff
@@ -1 +1,113 @@
-# Named
+### Named Types
+
+A named type provides a means to bind a symbolic name to a type
+and conforms to the 
+[named type](../../formats/model.md#3-named-type)
+in the super-structured data model.
+The named type [syntax](../../formats/sup.md#258-named-type)
+follows that of [SUP format](../../formats/sup.md), i.e.,
+a named type has the form
+```
+(<name>=<type>)
+```
+where `<name>` is an identifier or string and `<type>` is any type.
+
+Named types may be defined in four ways:
+* with a [`type`](../statements.md#type-statements) statement,
+* with a [cast](../expressions.md#casts),
+* with a definition inside of another type, or
+* by the input data itself.
+
+For example, this expression
+```
+80::(port=uint16)
+```
+casts the integer 80 to a named type called `port` whose type is `uint16`.
+
+Alternatively, named types can be declared with a type statement, e.g.,
+```
+type port = int16
+values 80::port
+```
+produces the value `80::(port=uint16)` as above.
+
+Type name definitions can be embedded in another type, e.g.,
+```
+type socket = {addr:ip,port:(port=uint16)}
+```
+defines a named type `socket` that is a record with field `addr` of type `ip`
+and field `port` of type `port`, where type `port` is a named type for type `uint16` .
+
+Named types may also be defined by the input data itself, as super-structured data is
+comprehensively self describing.
+When named types are defined in the input data, there is no need to declare their
+type in a query.
+In this case, a SuperSQL expression may refer to the type by the name that simply
+appears to the runtime as a side effect of operating upon the data.
+
+When the same name is bound to different types, a reference to that name is
+undefined except for the uses a definitions within a single heirarchical value,
+in which case, the most recent binding in depth-first order is used to resolve
+a reference to a type name.
```

Comment:
```suggestion
When the same name is bound to different types, a reference to that name is
undefined except for the definitions within a single hierarchical value,
in which case, the most recent binding in depth-first order is used to resolve
a reference to a type name.
```

I'm not 100% sure this is what you had in mind, but some fix was needed here beyond just the spelling of "hierarchical".